lib/modules/manager/composer/artifacts.ts

```diff
@@ -19,27 +19,44 @@ import {
 import { getRepoStatus } from '../../../util/git';
 import * as hostRules from '../../../util/host-rules';
 import { regEx } from '../../../util/regex';
+import { GitTagsDatasource } from '../../datasource/git-tags';
 import { PackagistDatasource } from '../../datasource/packagist';
 import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
 import type { AuthJson, ComposerLock } from './types';
 import {
   composerVersioningId,
   extractConstraints,
+  findGithubToken,
   getComposerArguments,
   getPhpConstraint,
+  isGithubPersonalAccessToken,
   requireComposerDependencyInstallation,
+  takePersonalAccessTokenIfPossible,
 } from './utils';
 
 function getAuthJson(): string | null {
   const authJson: AuthJson = {};
 
-  const githubCredentials = hostRules.find({
+  const githubToken = findGithubToken({
     hostType: PlatformId.Github,
     url: 'https://api.github.com/',
   });
-  if (githubCredentials?.token) {
+
+  const gitTagsGithubToken = findGithubToken({
+    hostType: GitTagsDatasource.id,
+    url: 'https://github.com',
+  });
+
+  const selectedGithubToken = takePersonalAccessTokenIfPossible(
+    githubToken,
+    gitTagsGithubToken
+  );
+  if (selectedGithubToken) {
+    if (!isGithubPersonalAccessToken(selectedGithubToken)) {
```

Comment:
The whole point of the original pull request is that github does not accept `ghs_` token anymore.

```
In Git.php line 471:
                                                                               
  Failed to execute git clone --mirror -- 'https://ghs...IQn:x-oauth-basic@gi  
  thub.com/kronostechnologies/account-service-sdk-php.git' '/tmp/renovate-cac  
  he/others/composer/vcs/git-github.com-kronostechnologies-account-service-sd  
  k-php.git/'                                                                  
                                                                               
  Cloning into bare repository '/tmp/renovate-cache/others/composer/vcs/git-g  
  ithub.com-kronostechnologies-account-service-sdk-php.git'...                 
  remote: Support for password authentication was removed on August 13, 2021.  
  remote: Please see https://docs.github.com/en/get-started/getting-started-w  
  ith-git/about-remote-repositories#cloning-with-https-urls for information o  
  n currently recommended modes of authentication.                             
  fatal: Authentication failed for 'https://github.com/kronostechnologies/acc  
  ount-service-sdk-php.git/'                   
  ```

Reply:
if GitHub won't accept app token, the hosted app would never work.

so it must be some other issue. maybe its because the token is passed as username instead as password. that seems to be the issue of the log above.

Reply:
> maybe its because the token is passed as username instead as password. 

Its the same for other applications. Username has to be set but is not used, i can be "Dummy" and Token is passed as password.

Reply:
sure, but the error above clearly has the opposite. the token is used as username.

Reply:
https://github.com/composer/composer/issues/11064

Reply:
That will be difficult to test without the risk of impacting users that had issues with the first fix https://github.com/renovatebot/renovate/issues/17778

Reply:
> It seems application token could work with this composer setting
> 
> ```
> {
>   "http-basic": {
>     "github.com": { 
>       "username": "x-access-token", 
>       "password": "ghs_XXXXX"
>     }
>   }
> }
> ```

so how would one use ghs_XXXXX

remove git-tags setting ?

Reply:
or how do i test the changes in this PR ? :)

Reply:
This is the official project documentation https://github.com/renovatebot/renovate/blob/main/docs/development/local-development.md

I found it difficult to test the exact case of a github application.

Reply:
@etremblay ty. but thats to much "Wall of Text" for me today :) 
ill keep waiting :)